pkg/controller/cluster/cluster_controller.go

```diff
@@ -135,11 +135,60 @@ func (r *ReconcileCluster) Reconcile(request reconcile.Request) (reconcile.Resul
 
 		// If no phase set default to pending for the initial phase
 		if instance.Status.Phase == "" {
+			instance.Spec.KopsConfig = CheckKopsDefaultConfig(instance.Spec)
+			// The following routine will remove any clusters from the state store that are not in etcd
+			// This will run whenever a cluster is created on the state store its beeing created in
+			reaper := []string{"REAPER"}
+			if utils.GetEnvs(reaper)[0][1] == "true" {
```

Comment:
Read Reaper from a config passed to ReconcileCluster. The ReconcileCluster will be instantiated with `{Reaper: bool}`

Reply:
@drewwells do you just mean storing the Reaper bool as a part of the ReconcileCluster object or are you talking about another way of passing in the value as well?

Reply:
Part of the object will work. Normally you have the constructor accept some parameters. A config object as the parameter is a good way to do this, then ReconcileCluster object would have this boolean in it

https://github.com/infobloxopen/cluster-operator/blob/656c7e7a2249daffaceb23dfbd50cf70ba7203f1/pkg/controller/cluster/cluster_controller.go#L42

```
type ReconcilerConfig struct {
  Mgr manager.Manager
  Reap bool
}

func NewReconciler(cfg ReconcilerConfig) reconcile.Reconciler {
...
```

Reply:
I've committed some changes to implement this. Please let me know if you see any issues